### PR TITLE
MySQL datetime guardrail

### PR DESCRIPTION
**Pull request by Tang8330**
If strict mode isn't enabled, MySQL may allow invalid datetime / timestamps.

### CODE REVIEW

File: lib/mysql/schema/convert.go [hunk 1]
```diff
@@ -82,13 +83,13 @@ func ConvertValue(value any, colType DataType) (any, error) {
 			return nil, fmt.Errorf("expected []byte got %T for value: %v", value, value)
 		}
 
-		if string(bytesValue) == "0000-00-00 00:00:00" {
-			// MySQL supports '0000-00-00 00:00:00' for datetime columns.
-			// We are returning `nil` here because this will fail most Time parsers.
+		stringValue := string(bytesValue)
+		if strings.HasSuffix(stringValue, "-00-00 00:00:00") {
```
Comment (nathan-artie) [inline lib/mysql/schema/convert.go:87]: Reading the MySQL docs it seems like it's also possible for the date to be non-zero and the month zero, or vice versa, so maybe what we want to do is lop off just the date, split by hyphens, and then check if any of the three segments is only zeros, something like:
```
func hasNonStrictModeDate(d string) bool {
	if len(d) < 10 {
		return false
	}
	parts := strings.Split(d[:10], "-")
	if len(parts) != 3 {
		return false
	}
	for _, part := range parts {
		value, err := strconv.Atoi(part)
		if err != nil {
			return false
		}
		if value == 0 {
			return true
		}
	}
	return false
}
```

Comment (Tang8330) [reply lib/mysql/schema/convert.go:87]: > date to be non-zero and the month zero

Oh god.

Comment (Tang8330) [reply lib/mysql/schema/convert.go]: Yeah makes sense

File: lib/mysql/schema/convert_test.go [hunk 2]
```diff
@@ -318,3 +318,13 @@ func TestConvertValues(t *testing.T) {
 		assert.Equal(t, []any{int64(1234), "hello world", true}, values)
 	}
 }
+
+func TestHasNonStrictModeDate(t *testing.T) {
+	assert.False(t, hasNonStrictModeDate("2021-01-02"))
```
Comment (nathan-artie) [inline lib/mysql/schema/convert_test.go:323]: I'd add a test for an empty string, a junk string, and a string with just two hyphens, all of which should return false.

File: lib/mysql/schema/convert.go (hunk 1 above)
Comment (nathan-artie) [reply lib/mysql/schema/convert.go]: Not to mention whether it's possible to have a zero day/month but a non-zero h/m/s. 😬

File: lib/mysql/schema/convert.go [hunk 3]
```diff
@@ -184,3 +185,27 @@ func ConvertValues(values []any, cols []Column) error {
 	}
 	return nil
 }
+
+// hasNonStrictModeDate - if strict mode is not enabled, we can end up having invalid datetimes
+func hasNonStrictModeDate(d string) bool {
```
Comment (nathan-artie) [inline lib/mysql/schema/convert.go:190]: Lol, I know I suggested this name, but maybe we could call this `hasNonStrictModeInvalidDate` to be a little clearer.
